### PR TITLE
ceph.spec.in: consolidate rhel+centos macros

### PR DESCRIPTION
**Pull request by smithfarm**
SUSE/openSUSE needs python-argparse and redhat-lsb-core is only
needed in RHEL/CentOS/Fedora.

http://tracker.ceph.com/issues/11638 Fixes: #11638

Signed-off-by: Nathan Cutler <ncutler@suse.cz>

### CODE REVIEW

File: ceph.spec.in [hunk 1]
```diff
@@ -144,7 +144,12 @@ Requires:	python-rados = %{epoch}:%{version}-%{release}
 Requires:	python-rbd = %{epoch}:%{version}-%{release}
 Requires:	python-cephfs = %{epoch}:%{version}-%{release}
 Requires:	python-requests
+%if 0%{defined suse_version}
+Requires:	python-argparse
+%endif
+%if 0%{?rhel} || 0%{?centos} || 0%{?fedora}
```
Comment (ktdreyer) [inline ceph.spec.in]: `0%{?rhel} || 0%{?centos}` can be simplified to just `0%{?rhel}`, since CentOS defines that macro for compatibility with RHEL.

**Comment by smithfarm** [comment]
@ktdreyer: ok, how about this then?

**Comment by ktdreyer** [comment]
This is really great. Thanks for merging the rhel+centos macros!

I'd prefer to see these changes as two separate commits: one that merges rhel + centos macros, and one that implements the OpenSUSE change for bzip2. It's easier to review when there's "one commit per change/one change per commit".

Please split them this into two commits and I'll merge.

**Comment by ktdreyer** [comment]
By the way, it's totally up to you of course, but it might be a good idea to add ncutler@suse.cz to your GitHub profile so we can look through the log of Ceph commits and see who's submitting them :)

**Comment by smithfarm** [comment]
@ktdreyer: OK, commit split into two. I added ncutler@suse.com (what my email address should have been set to the entire time) to my GitHub profile and set it as the public address. This does not match the signed-off-by line in these commits . . . sorry . . . I fixed my .gitconfig now as well.

**Comment by smithfarm** [comment]
BTW is my name shown in the log of Ceph commits now? (I can't verify because it displays fine for me)

**Comment by ktdreyer** [comment]
So if you look at one of the commits in this PR, say, https://github.com/SUSE/ceph/commit/46404dd381e240634d224e77adc69f0ba4807045 , you can see that it says "Nathan Cutler authored 20 minutes ago"... and "Nathan Cutler" really ought to be a link to your GitHub profile.

I think the problem is that the commits appear to be authored by suse.cz, not suse.com. Would you prefer to re-write them before I merge this, or just add the .cz address to GitHub as well?

**Comment by smithfarm** [comment]
Now I see what you meant. Fixed by adding ncutler@suse.cz to profile.

**Comment by ktdreyer** [comment]
Cool! merging...

**Pull request merged**